rs/rosetta-api/tests/system_tests/test_cases/search_transactions.rs

```diff
@@ -182,10 +206,214 @@ fn test_search_transactions_by_index() {
             },
         )
         .unwrap()
 }
+
+#[test]
+fn test_search_transactions_by_account() {
+    let mut runner = TestRunner::new(TestRunnerConfig {
+        max_shrink_iters: 0,
+        cases: *NUM_TEST_CASES,
+        ..Default::default()
+    });
+
+    runner
+        .run(
+            &(valid_transactions_strategy(
+                (*MINTING_IDENTITY).clone(),
+                DEFAULT_TRANSFER_FEE,
+                *MAX_NUM_GENERATED_BLOCKS,
+                SystemTime::now(),
+            )
+            .no_shrink()),
+            |args_with_caller| {
+                let rt = Runtime::new().unwrap();
+                rt.block_on(async {
+                    let env = RosettaTestingEnvironment::builder()
+                        .with_transfer_args_for_block_generating(args_with_caller.clone())
+                        .with_minting_account(MINTING_IDENTITY.sender().unwrap().into())
+                        .build()
+                        .await;
+                    let agent = get_test_agent(env.pocket_ic.url().unwrap().port().unwrap()).await;
+
+                    if !args_with_caller.is_empty() {
+                        let icp_blocks = query_encoded_blocks(&agent, u64::MAX, u64::MAX)
+                            .await
+                            .blocks
+                            .into_iter()
+                            .map(|block| icp_ledger::Block::decode(block).unwrap())
+                            .collect::<Vec<icp_ledger::Block>>();
+                        let account_identifiers = icp_blocks
+                            .clone()
+                            .iter()
+                            .flat_map(|block| match block.transaction.operation {
+                                icp_ledger::Operation::Transfer {
+                                    from, spender, to, ..
+                                } => spender
+                                    .map_or(vec![from, to], |spender| vec![from, to, spender]),
+                                icp_ledger::Operation::Mint { to, .. } => vec![to],
+                                icp_ledger::Operation::Burn { from, spender, .. } => {
+                                    spender.map_or(vec![from], |spender| vec![from, spender])
+                                }
+                                icp_ledger::Operation::Approve { from, spender, .. } => {
+                                    vec![from, spender]
+                                }
+                            })
+                            .collect::<HashSet<icp_ledger::AccountIdentifier>>();
+
+                        let mut search_transactions_request =
+                            SearchTransactionsRequest::builder(env.network_identifier.clone())
+                                .build();
+
+                        for account_id in account_identifiers.into_iter() {
+                            // We make sure that the service returns the correct number of transactions for each account
```

Comment:
I think checking that the correct number of transactions for each account is returned makes sense, but don't we here just check that any returned transactions contain the specified `account_id`? Wouldn't the test currently pass even if the `result` returned from `traverse_all_transactions` returned an empty vector?

Reply:
I'll add a check that makes sure it is not empty.